src/cluster/node.rs

```
@@ -25,3 +29,176 @@ pub fn read_node_id(p: &str) -> Result<String, ClusterError> {
         .map_err(|e| ClusterError::FailedReadingNodeID(e))?;
     Ok(contents)
 }
+
+/// Collection of all the metadata we can gather about the node. It is composed of
+/// sub-structs, listed below. This will be serialized to JSON and sent to Consul.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Metadata {
+    network: Option<NetworkMetadata>,
+    cpu: Option<CPUMetadata>,
+    ram: Option<RAMMetadata>,
+    disks: Vec<DiskMetadata>,
+    directories: Vec<DirectoryMetadata>,
+}
+
+impl Metadata {
+    pub fn gather() -> Metadata {
+        let sys = systemstat::System::new();
+        let cpu: Option<CPUMetadata>;
+        let ram: Option<RAMMetadata>;
+
+        if let Ok(cpu_metadata) = CPUMetadata::gather(&sys) {
+            cpu = Some(cpu_metadata);
+        } else {
+            cpu = None
+        }
+        if let Ok(ram_metadata) = RAMMetadata::gather(&sys) {
+            ram = Some(ram_metadata);
+        } else {
+            ram = None
+        }
+
+        Metadata {
+            network: None,
+            cpu: cpu,
+            ram: ram,
+            disks: Vec::new(),
+            directories: Vec::new(),
+        }
+    }
+}
+
+/// All network data about the node
+#[derive(Debug, Serialize, Deserialize)]
+pub struct NetworkMetadata {
+    ipv4: String,
+    ipv6: String,
+    port: u16,
+}
+
+/// CPU data about the node. Usage is meant to be 0.0-1.0. Not sure how to calculate it yet.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CPUMetadata {
+    // Key is "physical" or "logical"
+    // Value is how many of each the OS reports
+    physical: usize,
+    logical: usize,
+    five_min_load_average: Option<f32>,
+}
+
+impl CPUMetadata {
+    pub fn gather(sys: &systemstat::System) -> Result<CPUMetadata, ClusterError> {
+        let cpu = match sys.load_average() {
+            Ok(avg) => Some(avg.five),
+            Err(_) => None,
```

Why discard the error here?